install/install.sh

```diff
@@ -193,8 +193,28 @@ detect_platform() {
     cygwin_nt*) platform="pc-windows-msvc";;
     # mingw is Git-Bash
     mingw*) platform="pc-windows-msvc" ;;
-    # use the statically compiled musl bins on linux to avoid linking issues.
-    linux) platform="unknown-linux-musl" ;;
+    linux)
+      # default to the statically compiled musl bins on linux
+      # to avoid linking issues.
+      platform="unknown-linux-musl"
+      ldd_verstr="$(ldd --version | head -n1)"
+      ldd_glibc_prefix="ldd (GNU libc) "
+      if test "${ldd_verstr#*$ldd_glibc_prefix}" != "$ldd_verstr"; then
```

Comment:
Why `test` instead of `[]`?

Reply:
no particular reason, I can switch it back